cookbook/recipes/agent.rb

```diff
@@ -0,0 +1,55 @@
+include_recipe 'baragon::common'
+
+["#{node[:baragon][:proxy_conf_dir]}/proxy",
+ "#{node[:baragon][:upstream_conf_dir]}/upstreams"].each do |dir|
+  directory dir do
+    recursive true
+  end
+end
+
+creds = Chef::EncryptedDataBagItem.load('secrets',
+                                        'aws_credentials')['Baragon']
+
+baragon_agent_jar = 'BaragonAgentService-0.1.5-SNAPSHOT.jar'
+
+s3_file "/usr/share/java/#{baragon_agent_jar}" do
```

Comment:
Similar to Tom's comment on the aws databag, this seems very evertrue specific, we'd probably want to get the jar from a more public location. Maybe use the maven cookbook (https://github.com/opscode-cookbooks/maven) that way the maven repository is configurable if you ahve your own build of Baragon